sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala

```diff
@@ -539,3 +541,82 @@ case class StringToMap(text: Expression, pairDelim: Expression, keyValueDelim: E
 
   override def prettyName: String = "str_to_map"
 }
+
+/**
+ * Adds/replaces field in struct by name.
+ */
+case class WithFields(
+  structExpr: Expression,
+  nameExprs: Seq[Expression],
```

Comment:
can this be `Seq[String]`?

Reply:
Indeed, this can be changed to a `Seq[String]` easily. I did it as `Seq[Expression]` to make it easier in the future when/if we integrate this with the SQL API. I'm happy to change it if you think this isn't something to be concerned about today.

Reply:
fixed